src/System.Runtime.Extensions/tests/System/IO/PathTestsBase.cs

```diff
@@ -201,235 +201,253 @@ public partial class PathTestsBase
-            data.Add("foo", root + @"UNC\LOCALHOST\shareG", root + @"UNC\LOCALHOST\shareG\foo");
-            data.Add("foo", root + @"UNC\LOCALHOST\shareH\dir", root + @"UNC\LOCALHOST\shareH\dir\foo");
-            data.Add("foo", root + @"UNC\LOCALHOST\shareK\", root + @"UNC\LOCALHOST\shareK\foo");
-            data.Add("foo", root + @"UNC\LOCALHOST\  shareL\", root + @"UNC\LOCALHOST\  shareL\foo");
-
-            // Local host paths
-            data.Add(@"foo", @"\\server1", @"\\server1\foo");
-            data.Add(@"\foo", @"\\server2", @"\\server2\foo");
-            data.Add(@"foo", @"\\server3\", @"\\server3\foo");
-            data.Add(@"foo", @"\\server3\\", @"\\server3\foo");
-            data.Add(@"..\foo", @"\\server4", @"\\server4\..\foo");
-            data.Add(@".\foo", @"\\server5\share", @"\\server5\share\foo");
-            data.Add(@"..\foo", @"\\server6\share", @"\\server6\share\foo");
-            data.Add(@"\foo", @"\\a\b\\", @"\\a\b\foo");
-
-            data.Add(@"foo", @"\\LOCALHOST\share8\test.txt.~SS", @"\\LOCALHOST\share8\test.txt.~SS\foo");
-            data.Add(@"foo", @"\\LOCALHOST\share9", @"\\LOCALHOST\share9\foo");
-            data.Add(@"foo", @"\\LOCALHOST\shareA\dir", @"\\LOCALHOST\shareA\dir\foo");
-            data.Add(@". \foo", @"\\LOCALHOST\shareB\", @"\\LOCALHOST\shareB\. \foo");
-            data.Add(@".. \foo", @"\\LOCALHOST\shareC\", @"\\LOCALHOST\shareC\.. \foo");
-            data.Add(@"    \foo", @"\\LOCALHOST\shareD\", @"\\LOCALHOST\shareD\    \foo");
-
-            data.Add("foo", @"\\LOCALHOST\share1", @"\\LOCALHOST\share1\foo");
-            data.Add("foo", @"\\LOCALHOST\share2", @"\\LOCALHOST\share2\foo");
-            data.Add("foo", @"\\LOCALHOST\share3\dir", @"\\LOCALHOST\share3\dir\foo");
-            data.Add("foo", @"\\LOCALHOST\share4", @"\\LOCALHOST\share4\foo");
-            data.Add("foo", @"\\LOCALHOST\share5", @"\\LOCALHOST\share5\foo");
-            data.Add("foo", @"\\LOCALHOST\share6\", @"\\LOCALHOST\share6\foo");
-            data.Add("foo", @"\\LOCALHOST\  share7\", @"\\LOCALHOST\  share7\foo");
-
-            return data;
-        }
+            // UNC path starting with \\?\UNC\
+            { @"foo", @"\\?\UNC\", @"\\?\UNC\foo" },
+            { @"foo", @"\\?\UNC\server1", @"\\?\UNC\server1\foo" },
+            { @"\foo", @"\\?\UNC\server2", @"\\?\UNC\server2\foo" },
+            { @"foo", @"\\?\UNC\server3\", @"\\?\UNC\server3\foo" },
+            { @"foo", @"\\?\UNC\server3\\", @"\\?\UNC\server3\\foo" },
+            { @"..\foo", @"\\?\UNC\server4", @"\\?\UNC\server4\..\foo" },
+            { @".\foo", @"\\?\UNC\server5\share", @"\\?\UNC\server5\share\foo" },
+            { @"..\foo", @"\\?\UNC\server6\share", @"\\?\UNC\server6\foo" },
+            { @"\foo", @"\\?\UNC\a\b\\", @"\\?\UNC\a\b\foo" },            
+            { @"foo", @"\\?\UNC\LOCALHOST\share8\test.txt.~SS", @"\\?\UNC\LOCALHOST\share8\test.txt.~SS\foo" },
```

Comment:
This is getting the same exact result as `\\.\UNC\` and `\\`.  Same seems to be true for most of these. Please pull these into a shared test case. Like this:

``` C#
// ...
{ @"foo", @"", @"foo" },
{ @"foo", @"LOCALHOST\share8\test.txt.~SS", @"LOCALHOST\share8\test.txt.~SS\foo" },
// ...

public void GetFullPath_CommonUnc_Windows(string path, string basePath, string expected)
{
     Assert.Equal(@"\\" + expected, Path.GetFullPath(path, @"\\" + basePath));
     Assert.Equal(@"\\.\UNC\" + expected, Path.GetFullPath(path, @"\\.\UNC\" + basePath));
     Assert.Equal(@"\\?\UNC\" + expected, Path.GetFullPath(path, @"\\?\UNC\" + basePath));
}
```

If you iterate like this I'm sure you'll find other commonalities between the sets of input data. Same sort of thing should be done for non-rooted normal paths:

``` C#
public void GetFullPath_CommonUnrooted_Windows(string path, string basePath, string expected)
{
     Assert.Equal(expected, Path.GetFullPath(path, basePath));
     Assert.Equal(@"\\.\" + expected, Path.GetFullPath(path, @"\\.\" + basePath));
     Assert.Equal(@"\\?\" + expected, Path.GetFullPath(path, @"\\?\" + basePath));
}
```